frontend/src/libs/components/meditation-timer/meditation-timer.tsx

```diff
@@ -0,0 +1,69 @@
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback, useState } from '#libs/hooks/hooks.js';
+
+import { DURATION_UNIT, MEDITATION_DURATION } from './constants/constants.js';
+import styles from './styles.module.scss';
+
+type TimerProperties = {
+  onClose: () => void;
+  defaultDuration: string;
+};
+
+const MeditationTimer: React.FC<TimerProperties> = ({
+  onClose,
+  defaultDuration,
+}) => {
+  const [selectedDuration, setSelectedDuration] = useState(defaultDuration);
+
+  const handleClick = useCallback((duration: string) => {
+    return () => {
+      setSelectedDuration(duration);
+    };
+  }, []);
+
+  return (
+    <div className={styles['timer']}>
+      <p className={styles['title']}>Choose Your Duration</p>
+      <div className={styles['duration-container']}>
+        {Object.keys(MEDITATION_DURATION).map((duration) => {
+          return (
+            <button
```

Comment:
I think you can try to use common `Checkbox` component because it provides `style:checked` option. At least my suggested changes below allows you to create and use new preseted stye like on the common `Button` coponent. 

Also I suppose it's needed to implement that choices validation via `useAppForm`, you'll be able to reset it `onChange` and then new checkbox will be selected automatically

Reply:
I defo think you have to use `useAppForm`, because in the similar situation in survey page I was asked to do so, you can check how it is implemented there in the `PreferencesStep` component. 

About `Ckeckbox` component insetad of `Button`, it's just more obvious and easier to handle selectable options with it